blackboxopt/base.py

```diff
@@ -163,11 +180,15 @@ def __init__(
         _raise_on_duplicate_objective_names(objectives)
         self.objectives = objectives
 
-    def report_evaluation(self, evaluation: Evaluation) -> None:
-        raise_on_unknown_or_incomplete(
-            exception=ObjectivesError,
-            known=[o.name for o in self.objectives],
-            reported=evaluation.objectives.keys(),
-        )
+    def report(self, evaluations: Union[Evaluation, Iterable[Evaluation]]) -> None:
+        if isinstance(evaluations, Evaluation):
+            evaluations = [evaluations]
+
+        for evaluation in evaluations:
+            raise_on_unknown_or_incomplete(
+                exception=ObjectivesError,
+                known=[o.name for o in self.objectives],
+                reported=evaluation.objectives.keys(),
+            )
```

Comment:
It seems to me we are being inconsistent with the different failure modes for evaluations. In case of objectives errors we don't accept any evaluations if one is off. In case of the missing IDs in BOHB, we accept all evaluations with IDs and only reject / raise on the ones that have issues.

Reply:
True, what do you propose?

Reply:
I'd propose to follow what we decided in the previous discussion, to accept all evaluations that don't fail and then raise on the failed ones. Which links to [this comment](https://github.com/boschresearch/blackboxopt/pull/19/files#r704079665).